dashboard/app/controllers/reports_controller.rb

```diff
@@ -56,6 +56,32 @@ def all_usage
     render 'usage', formats: [:html]
   end
 
+  def search_for_teachers
+    authorize! :read, :reports
+    address_filter = params[:addressFilter]
+    email_filter = params[:emailFilter]
+    fields = ['id', 'name', 'email', 'num_students']
+    teachers = User.find_by_sql(<<SQL)
+SELECT
+  u.id, u.name, u.email, COUNT(f.student_user_id) AS num_students
+FROM (
+  SELECT id, name, email
+  FROM users
+  WHERE user_type = 'teacher'
+    AND email LIKE '%#{email_filter}%'
+    AND full_address LIKE '%#{address_filter}%'
+) AS u
+LEFT JOIN (
+  SELECT user_id, student_user_id
+  FROM followers
+) AS f
+ON u.id = f.user_id
+GROUP BY u.id, u.name, u.email
+SQL
```

Comment:
AFAIK, it could be done through the ActiveRecord ORM or through raw SQL. Given that this query will only get more complicated in future iterations, I see the SQL as being more readable. But if the team disagrees, I'm happy to convert it to ActiveRecord.